athena-jdbc/src/main/java/com/amazonaws/connectors/athena/jdbc/snowflake/SnowflakeMetadataHandler.java

```diff
@@ -0,0 +1,169 @@
+{
+    static final Map<String, String> JDBC_PROPERTIES = ImmutableMap.of("databaseTerm", "SCHEMA");
+    private static final Logger LOGGER = LoggerFactory.getLogger(SnowflakeMetadataHandler.class);
+
+    /**
+     * Instantiates handler to be used by Lambda function directly.
+     *
+     * Recommend using {@link com.amazonaws.connectors.athena.jdbc.MultiplexingJdbcCompositeHandler} instead.
+     */
+    public SnowflakeMetadataHandler()
+    {
+        this(JDBCUtil.getSingleDatabaseConfigFromEnv(JdbcConnectionFactory.DatabaseEngine.SNOWFLAKE));
+    }
+
+    public SnowflakeMetadataHandler(final DatabaseConnectionConfig databaseConnectionConfig)
+    {
+        super(databaseConnectionConfig, new GenericJdbcConnectionFactory(databaseConnectionConfig, JDBC_PROPERTIES));
+    }
+
+    @VisibleForTesting
+    protected SnowflakeMetadataHandler(final DatabaseConnectionConfig databaseConnectionConfig, final AWSSecretsManager secretsManager,
+            final AmazonAthena athena, final JdbcConnectionFactory jdbcConnectionFactory)
+    {
+        super(databaseConnectionConfig, secretsManager, athena, jdbcConnectionFactory);
+    }
+
+    @Override
+    protected String escapeNamePattern(final String name, final String escape)
+    {
+        return super.escapeNamePattern(name, escape).toUpperCase();
+    }
+
+    @Override
+    protected ResultSet getColumns(final String catalogName, final TableName tableHandle, final DatabaseMetaData metadata)
+            throws SQLException
+    {
+        String escape = metadata.getSearchStringEscape();
+        LOGGER.warn(
+          "[SNOWFLAKE DEBUG] catalogName: {}, SchemaName: {}, TableName: {}",
+          catalogName,
+          escapeNamePattern(tableHandle.getSchemaName(), escape),
+          escapeNamePattern(tableHandle.getTableName(), escape)
+        );
+        return metadata.getColumns(
+                catalogName,
+                escapeNamePattern(tableHandle.getSchemaName(), escape),
+                escapeNamePattern(tableHandle.getTableName(), escape),
+                null);
+    }
+
+    @Override
+    public Schema getPartitionSchema(final String catalogName)
+    {
+        LOGGER.warn("[SNOWFLAKE DEBUG] getPartitionSchema ");
+        SchemaBuilder schemaBuilder = SchemaBuilder.newBuilder();
+        return schemaBuilder.build();
+    }
+
+    @Override
+    public void getPartitions(final BlockWriter blockWriter, final GetTableLayoutRequest getTableLayoutRequest, QueryStatusChecker queryStatusChecker)
+    { }
+
+    @Override
+    public GetSplitsResponse doGetSplits(BlockAllocator blockAllocator, GetSplitsRequest getSplitsRequest)
+    {
+        LOGGER.info("doGetSplits: enter - " + getSplitsRequest);
+
+        String catalogName = getSplitsRequest.getCatalogName();
+        Set<Split> splits = new HashSet<>();
+
+        Split split = Split.newBuilder(makeSpillLocation(getSplitsRequest), makeEncryptionKey()).build();
+        splits.add(split);
```

Comment:
Does snowflake support any partitioning or parallelism that we can exploit to create multiple splits?

Reply:
That's the big question, Snowflake supports micro-partitioning but not sure yet how that would fit in the structure of the code yet (https://docs.snowflake.com/en/user-guide/tables-clustering-keys.html)

I'm open to feedback on that